### PR TITLE
fix #7120 chore(nimbus): wrap all JEXL expressions in parens

### LINKED ISSUE #7120: Wrap targeting config JEXL in outer parens
We always want to treat a targeting configs expression as an atomic unit, so we should wrap it in outer parens here:

https://github.com/mozilla/experimenter/blob/main/app/experimenter/experiments/models/nimbus.py#L250

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-2271)

### PR DESCRIPTION
**Pull request by jaredlockhart**
Because

* Arbitrary JEXL expressions can contain arbitrary subclauses
* To ensure the JEXL evaluator groups them the way is intended by the subclause authors

This commit

* Wraps all JEXL subclauses in outer parens

### CODE REVIEW

File: app/experimenter/experiments/models/nimbus.py [hunk 1]
```diff
@@ -296,7 +296,7 @@ def targeting(self):
         if len(expressions) == 0:
             return "true"
 
-        return " && ".join(expressions)
+        return " && ".join([f"({expression})" for expression in expressions])
```
Comment (eliserichards) [inline app/experimenter/experiments/models/nimbus.py:299]: 👏

File: app/experimenter/experiments/tests/api/v6/test_serializers.py [hunk 2]
```diff
@@ -54,9 +54,9 @@ def test_expected_schema_with_desktop_single_feature(self):
                 # DRF manually replaces the isoformat suffix so we have to do the same
                 "startDate": experiment.start_date.isoformat().replace("+00:00", "Z"),
                 "targeting": (
-                    'browserSettings.update.channel == "nightly" '
-                    "&& version|versionCompare('94.!') >= 0 "
-                    "&& 'app.shield.optoutstudies.enabled'|preferenceValue"
```
Comment (eliserichards) [inline app/experimenter/experiments/tests/api/v6/test_serializers.py:59]: Should we be testing both cases (with and without parens)?

**Review by eliserichards — APPROVED** [review]
Looks good! Just one q

File: app/experimenter/experiments/tests/api/v6/test_serializers.py (hunk 2 above)
Comment (jaredlockhart) [reply app/experimenter/experiments/tests/api/v6/test_serializers.py:59]: No no there won't be a non paren case anymore

**Pull request merged**